src/import_edf.jl

```diff
@@ -51,14 +41,21 @@ function _normalize_references(original_label, canonical_names)
     return first(parts), recombined
 end
 
+_safe_lowercase(c::Char) = isvalid(c) ? lowercase(c) : c
+
+# malformed UTF-8 chars are a choking hazard
+_safe_lowercase(s::AbstractString) = map(_safe_lowercase, s)
+
 function match_edf_label(label, signal_names, channel_name, canonical_names)
-    type, spec = edf_type_and_spec(lowercase(label))
-    if spec === nothing
-        label = type
-    else
-        any(==(type), signal_names) || return nothing
-        label = spec
+    label = _safe_lowercase(label)
+    for signal_name in signal_names
+        m = match(Regex("[\\s\\[,\\]]*$(signal_name)[\\s,\\]]*\\s+(?<spec>.+)", "i"), label)
```

Comment:
This one heck of a regex... Could you add a comment above it or something giving an example of what you expect it to match?